include/simdjson/generic/ondemand/json_iterator-inl.h

```diff
@@ -191,6 +191,14 @@ inline std::string json_iterator::to_string() const noexcept {
           + std::string(" ]");
 }
 
+inline simdjson_result<const char *> json_iterator::current_location() noexcept {
+  auto cur_struct_index = token.position() - parser->implementation->structural_indexes.get();
```

Comment:
It is possible for parser to be null if `is_alive()` is false, so for safety, you should check whether parser is null or not. But actually, it bring up a problem because if `abandon` is called, then your function won't work since we have made parse null!!!